Initial seedtime limit based on tracker or label

vanossj opened the pull request:
I added a support for setting the initial seedtime limit based on tracker or label. The user can set up a table of regex filters in the gtkui.

![preferencesscreen](https://cloud.githubusercontent.com/assets/8310169/6694328/f789d5e6-ccaf-11e4-9bf8-a8e759554453.PNG)

I would have liked to add all filter code to post_torrent_add hook, but I could not see a way to guarantee that the label post_torrent_add hook would be processed first. So instead I deferred the call by 1 second.

Suggestions welcome

vanossj committed:
Added support for setting the stop seed time on torrent add based on a table of filters
filters can be based on label or tracker information

vanossj committed:
pep8 code cleanup

vanossj committed:
updated glade file to use stock buttons so that themes that use icons will see them

Symbiot78 commented:
Hi
this is great work and exactly what I need... 
how do I implement it?

vanossj commented:
Can you clarify what you mean by 'implement it'?
Do you want to build the plugin? install it? use the plugin?

Symbiot78 commented:
Hi

I really just want to use it :-) 
I tried building it and I MIGHT have it installed.. but I am unsure on the proper usage.. perhaps if you gave a short usage guide I can give it a go :-)

Symbiot78 commented:
Ok, been playing a little with what I think is your build.. I created a label called 'test'. I set the field to "label" & Filter to "test" & seedtime to 0..
but when the torrent is added is it given a 3 day seedtime no matter what.. I cannot set a label when the torrent is loaded so I am thinking it sets the default seed time rule and then doesn't change it ? 

![1](https://cloud.githubusercontent.com/assets/3314249/14020095/d0768746-f1d5-11e5-9480-13ac378c646d.PNG)
![2](https://cloud.githubusercontent.com/assets/3314249/14020096/d076a000-f1d5-11e5-8633-ea3440d4cc8b.PNG)

Symbiot78 commented:
oh this is for manually added torrents.. I haven't tested on autoadded torrents..

vanossj committed:
Create README.md

Symbiot78 commented:
Ok, it works for autoadded torrents :-)

vanossj commented:
I updated my README and added a release, but it looks like you've gotten that far already.

The seed time is filters on information available when a torrent is added, manually or automatically. Filtering on tracker works all the time because that information is always there when a torrent is added. The caveat for label filters is that there is not a way to specify a label when adding a torrent manually via the gui. So these torrents don't yet have a label when added and that's why they are falling through to the default filter.

There are some possibly workarounds. The plugin could wait 1 minute before assigning a seed time. This way, if you changed the label right after you add the torrent, the seedtime filters would see that label and assign the expected stopping time

Symbiot78 commented:
Hi
setting a delay would be pretty good.. a few minutes.. or even better... the OPTION to set a delay manually in the plugin..
I set up a simple seedtime filter as in the picture.. I sorted it by importance and used both label & tracker to split torrents that have different labels but can potentially be from the same tracker..

see screenshot..
![3](https://cloud.githubusercontent.com/assets/3314249/14021101/4f1e8e32-f1da-11e5-8efa-633b0d71978c.PNG)

Symbiot78 commented:
Could the plugin also be updated so in main view under the column "stop seed time" you could actually see the remaining seed time before 'seedtime' plugin removes the torrent? it only shows the initial time and doesn't change..

vanossj committed:
Made the initial delay configurable. This gives user time to change the label before the stop seed time is assigned

vanossj committed:
set delay time in gui to value from config file

vanossj commented:
added the configurable delay.

There is already "stop seed time" and a "seeding time" columns, a remaining seed time would just be the difference between the two. Would you use a "remaining seed time" column instead of the other two? Trying to strike a balance between useful options and too many options.

Symbiot78 commented:
Hi

Great work.. will be testing soon.. !

I'd personally prefer to have a "remaining seed time" column since all my torrent are using the 'seedtime' plugin. Just having a static column like it is now is of no use.. only thing you can use if for is to verify that the seedtime has been set.. which would be apparent if the 'remaining seedtime' column is present.
I assume you would be able to 'just' add the column and let the user choose? 
I'm not a programmer so I haven't got a clue as to the level of difficulty in doing this.. 

I would think it a great feature to add to the plugin ..

vanossj committed:
Added seed time remaining column

vanossj commented:
There you go

Symbiot78 commented:
This is versuching fantastic..! great work Vanossj...! The seedtime plugin just got way way way more useful to me...  I hope more people will find this plugin!!

![seedtime](https://cloud.githubusercontent.com/assets/3314249/14067165/74634940-f45e-11e5-9911-8905f70143f0.png)

vanossj committed:
added beginnings of seedtime webui

vanossj committed:
added grid view for filters

vanossj committed:
added grid panel and buttons

vanossj committed:
insert new filter on top

vanossj committed:
switch buttons to gridpanel, added todo list

vanossj committed:
added up/down/add/remove functionality

vanossj committed:
hard coded height, until something better can be figured out

vanossj committed:
added cell editing

vanossj committed:
added save configuration, still need to get field to save correctly

vanossj committed:
Added status columns, added context menu

vanossj committed:
added custom stop time window

vanossj committed:
added time format for seedtime columns

vanossj committed:
format time 0 as 0 not infinate

vanossj committed:
added extjs combobox for field selection

vanossj committed:
WIP: started removing default from filterlist

vanossj committed:
removed default from filter table

vanossj committed:
added default seed time to gtk

vanossj committed:
bug fix: added ,

vanossj committed:
added decimal precision to default stop time

vanossj committed:
removed completed todo

vanossj committed:
Fixed webui bug, should use stop_time, not stoptime in order to match core

vanossj committed:
versioning seems like a good idea

vanossj committed:
use correct version for 2.0.0 release

NHellFire committed:
Support for Deluge 2.0

vanossj committed:
Merge pull request #6 from NHellFire/master

Support for Deluge 2.0

vanossj committed:
working on deluge 2
